File: 2periodo/TrabalhosPraticos/2024-1/TP1-Aeds2/AlteracaoAleatoria/Aleatoria.java
import java.util.Random;
/*
*   806454 - Yago Almeida Melo
*   TP1 - Alteracao Aleatoria em Java - 2024/1
*/

public class Aleatoria {   
    static Random gerador = new Random();
    
     /*
     * function: isFim
     * @params: String 
     * action: Recebe uma String e retorna verdadeiro se a String for igual a "FIM", senão retorna falso.
     */
    public static boolean isFim(String input){
        return input.charAt(0) == 'F' && input.charAt(1) == 'I' && input.charAt(2) == 'M';
    }

    /*
     * function: AlteracaoAleatoria
     * @params: String
     * action: Sorteia uma letra de 'a' a 'z' para ser trocada com outra letra sorteada em uma String
     * e retorna esta String modificada. 
     */
    public static String AlteracaoAleatoria(String input){
        String resp = "";
        char sorteado1 = (char) ('a'+(Math.abs(gerador.nextInt()) % 26));
        char sorteado2 = (char) ('a'+(Math.abs(gerador.nextInt()) % 26));
        for(int i=0;i<input.length();i++){
            if(input.charAt(i)==sorteado1){
                resp += sorteado2;
            } else{
                resp += input.charAt(i);
            }
        }
        return resp;
    }
    public static void main(String[] args){
        gerador.setSeed(4);
        String input = MyIO.readLine(); 
        while(!isFim(input)){
            MyIO.println(AlteracaoAleatoria(input));
            input = MyIO.readLine(); 
        }
    }
}

File: 2periodo/TrabalhosPraticos/2024-1/TP1-Aeds2/AlteracaoAleatoria/aleatoria.c
/*
    806454 - Yago Alemida Melo
    TP-1:2024/1  - Alteração aleatoria em C
*/

#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <stdbool.h>

bool isFim(char ptr[]);
char * alteracaoAleatoria(char ptr[]);
char * alteracao(char ptr[], int i);

/*
*   function: isFim
*   @params: char[]
*   action: Retorna verdadeiro se string é igual a "FIM" e falso se não for.
*/
bool isFim(char ptr[]){
    return strlen(ptr) == 4 && ptr[0] == 'F' && ptr[1] == 'I' && ptr[2] == 'M' && ptr[3] == '\n';
}

/*
*   function: alteracaoAleatoria
*   @params: char[]
*   action: chamada para funcao recursiva alteracao e retorna a string modificada
*/
char * alteracaoAleatoria(char ptr[]){
    return alteracao(ptr, 0);
     
}

/*

*   function: alteracao
*   @params: char[] & int
*   action: funcao recursiva que faz a alteracao e retorna a string modificada
*/
char * alteracao(char ptr[], int i){
    char atual = ('a' + rand() % 26);
    char nova = ('a' + rand() % 26);
    if(i < strlen(ptr)){
        if(ptr[i] == atual){
            ptr[i] = nova;
        }
        alteracao(ptr, i+1);
    }
    return ptr;
}    

int main(){
    char input[1000];
    srand(4);
    fgets(input, sizeof(input), stdin);
    while(!isFim(input)){
        char *resp = alteracaoAleatoria(input);
        printf("%s", resp);
        fgets(input, sizeof(input), stdin);
    }
    return 0;
}